exporters/otlp/src/otlp_grpc_client.cc

```diff
@@ -565,21 +655,39 @@ bool OtlpGrpcClient::ForceFlush(std::chrono::microseconds timeout) noexcept
   }
 
   return timeout_steady > std::chrono::steady_clock::duration::zero();
+#else
+  return true;
+#endif
 }
 
-bool OtlpGrpcClient::Shutdown(std::chrono::microseconds timeout) noexcept
+bool OtlpGrpcClient::Shutdown(OtlpGrpcClientReferenceGuard &guard,
+                              OPENTELEMETRY_MAYBE_UNUSED std::chrono::microseconds timeout) noexcept
 {
   if (!async_data_)
   {
     return true;
   }
 
+  bool last_reference_removed = RemoveReference(guard);
   bool force_flush_result;
-  if (false == is_shutdown_.exchange(true, std::memory_order_acq_rel))
+  if (last_reference_removed && false == is_shutdown_.exchange(true, std::memory_order_acq_rel))
   {
     force_flush_result = ForceFlush(timeout);
 
-    async_data_->cq.Shutdown();
+#ifdef ENABLE_ASYNC_EXPORT
+    std::unordered_set<std::shared_ptr<OtlpGrpcAsyncCallDataBase>> running_calls;
+    {
+      std::lock_guard<std::mutex> lock(async_data_->running_calls_lock);
+      running_calls = async_data_->running_calls;
```

Comment:
Are all the call_data moved to running_calls? If not, all the pending data may be canceled again in the dtor `~OtlpGrpcLient`.

Reply:
Yes, and the owner of OtlpGrpcClient will call Shutdown before destructor, which will call ForceFlush first to flush pending datas. The destructor of OtlpGrpcClient is used to avoid indefinite waits when some problems happens.